warehouse/search/__init__.py

```diff
@@ -18,7 +18,32 @@
 from celery.schedules import crontab
 from elasticsearch_dsl import serializer
 
+from warehouse import db
 from warehouse.search.utils import get_index
+from warehouse.packaging.models import Project, Release
+
+
+@db.listens_for(db.Session, "after_flush")
+def store_projects_for_project_reindex(config, session, flush_context):
+    # We'll (ab)use the session.info dictionary to store a list of pending
+    # Project updates to the session.
+    purges = session.info.setdefault("warehouse.search.project_updates", set())
```

Comment:
Probably want to call this something other than `purges`... `reindexable`? `changed_projects`? `updated_projects`?